internal/core/src/segcore/SegmentSealedImpl.cpp

```diff
@@ -1517,6 +1520,14 @@ SegmentSealedImpl::generate_binlog_index(const FieldId field_id) {
                                    field_index_meta,
                                    segcore_config_,
                                    SegmentType::Sealed));
+            if (row_count < field_binlog_config->GetBuildThreshold()) {
```

Comment:
It is a common case, and the condition rule is used to reduce useless binlog index generation. It could be better to process it as a normal case and set the log level with `info`.